### PR TITLE
docs: get started docs r-r version

### PR DESCRIPTION
**Pull request by adnjoo**
### Summary

edit `get-started.md` to include latest version of react-rails #1324

### Pull Request checklist
_Remove this line after checking all the items here. If the item is not applicable to the PR, both check it out and wrap it by `~`._

~- [] Add/update test to cover these changes~
- [x] Update documentation
~- [ ] Update CHANGELOG file~

### CODE REVIEW

File: docs/get-started.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ Also update the Babel configuration in the `package.json` file:
 
 4. Install `react-rails`:
 ```bash
-$ bundle add 'react-rails' --strict
+$ bundle add 'react-rails' --strict --version '3.1.1'
```
Comment (ahangarha) [inline docs/get-started.md:54]: This way, we should update the docs for every single release. Isn't it?

Comment (adnjoo) [reply docs/get-started.md:54]: hmm how about like `bundle add 'react-rails' --strict --version '>=3.1.1'`

Comment (ahangarha) [reply docs/get-started.md]: It is important to install "exact" version for both the Ruby gem and node package. This is emphasized in the documentation as well.

File: docs/get-started.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ Also update the Babel configuration in the `package.json` file:
 
 4. Install `react-rails`:
 ```bash
-$ bundle add 'react-rails' --strict
+$ bundle add 'react-rails' --strict --version '>=3.1.1'
```
Comment (ahangarha) [inline docs/get-started.md:54]: We should use "exact" version.

**Review by ahangarha — CHANGES_REQUESTED** [review]

File: docs/get-started.md (hunk 2 above)
Comment (ahangarha) [reply docs/get-started.md:54]: But I see one point. `--strict` doesn't count the patch version and when we have fixes and bump the patch value in the version, `--strict` doesn't help.

If there is any better proposal (than manually fix the version) please share.

**Comment by G-Rath** [comment]
Is there a reason why we're suggesting using `--strict` in the first place?

**Comment by ahangarha** [comment]
> Is there a reason why we're suggesting using `--strict` in the first place?

Because `bundle add react-rails` adds `gem "react-rails", "~> 3.1"` which means any 3.x version of react-rails`. Since `react-rails` has both Ruby and Node sides, we need to ensure we use "exact" and the same versions for both.

Any suggestion?

**Comment by G-Rath** [comment]
ah right it's `react_ujs` and it's installed by `rails generate react:install` which is why it's not listed in the getting started docs.

As I've pushed for over in Shakapacker, I don't think we should be encouraging people to pin to exact versions since it tends to make things more brittle in managing updates (especially for security fixes), and the whole ideal of patch versions is they should not be breaking - I know it's not impossible for there to be a dependency between the Ruby and JS versions of the package for a particular bug fix, but in reality how often has that actually happened?

I think the better long-term path is to help the user ensure the versions are in sync like what's being done in Shakapacker - effectively https://github.com/shakacode/shakapacker/issues/156, which probably could be done by just copying over the version detection logic stuff? Noting though I have actually somewhat deliberately ignored this part of `react-rails` for the time being to focus on the `package_json` work because unlike with Shakapacker I have not run into any version mismatch issues and I think it might make sense for some of the JS version stuff to actually live in `package_json`, but don't want to rush into it and end up with a poor design.

tbh it feels like a bug in Bundler that `--strict` does not give you the latest patch version, given there's no other way to dynamically resolve the latest version with `--strict`...

**Comment by ahangarha** [comment]
> tbh it feels like a bug in Bundler that --strict does not give you the latest patch version, given there's no other way to dynamically resolve the latest version with --strict...

I agree. Twice, I thought about it but then left the issue. Perhaps they cannot change the behavior of the `strict` option but maybe can add `--strict-patch` option. Strict should mean STRICT!!! But... 😄

**Comment by justin808** [comment]
@ahangarha @G-Rath ready to merge?

File: docs/get-started.md (hunk 2 above)
Comment (justin808) [reply docs/get-started.md:54]: I'm not seeing the value of any changes. What am I missing?

**Comment by justin808** [comment]
If this changes, then we should also change react_on_rails for the same issue. 

We have this documentation: https://www.shakacode.com/react-on-rails/docs/getting-started/

**Comment by ahangarha** [comment]
By the way, thanks for the initiative.
@adnjoo

**Comment by adnjoo** [comment]
@ahangarha np, thanks for reviewing :)

**Pull request closed without merging**